lib/Fmt_ast.ml

```diff
@@ -396,20 +400,23 @@ let docstring_epi ~standalone ~next ~epi ~floating =
       str "\n" $ epi
   | _ -> epi
 
-let fmt_docstring c ?(standalone = false) ?pro ?epi doc =
-  list_pn (Option.value ~default:[] doc)
-    (fun ~prev:_ ({txt; loc}, floating) ~next ->
+let fmt_docstring' c ?(standalone = false) ?pro ?epi doc =
+  list_pn doc (fun ~prev:_ ({txt; loc}, floating) ~next ->
       let epi = docstring_epi ~standalone ~next ~epi ~floating in
       fmt_parsed_docstring c ~loc ?pro ~epi txt (Docstring.parse ~loc txt) )
 
+let fmt_docstring c ?standalone ?pro ?epi doc =
+  fmt_docstring' c ?standalone ?pro ?epi (Option.value ~default:[] doc)
+
+(** Accept up to two docstrings, in reverse order of appearance. *)
 let fmt_docstring_around_item' ?(is_val = false) ?(force_before = false)
-    ?(fit = false) c doc1 doc2 =
-  match (doc1, doc2) with
-  | Some _, Some _ ->
-      ( fmt_docstring c ~epi:(fmt "@\n") doc1
-      , fmt_docstring c ~pro:(fmt "@\n") doc2 )
-  | None, None -> (noop, noop)
-  | None, Some doc | Some doc, None -> (
+    ?(fit = false) c docs =
```

Comment:
The new signature is very unsafe. I would prefer to keep the previous one. `extract_doc_attrs` should be the only place these values are produced and already does the required logic.

Reply:
I tried to change it, but it a bit unwieldy for fmt_docstring_around_item_attrs lower. I'm fine either way, and I think you wrote this code, it should easier for you to change it.

Reply:
I'm preparing a patch.

Reply:
I've pushed this directly to your branch. The change in docstring formatting is reverted.